cob_twist_controller/include/cob_twist_controller/kinematic_extensions/kinematic_extension.h

```diff
@@ -0,0 +1,128 @@
+#include <tf/tf.h>
+
+#include <Eigen/Geometry>
+
+
+
+/* BEGIN KinematicExtensionBuilder *****************************************************************************************/
+/// Class providing a static method to create kinematic extension objects.
+class KinematicExtensionBuilder
+{
+    public:
+        KinematicExtensionBuilder() {}
+        ~KinematicExtensionBuilder() {}
+        
+        static KinematicExtensionBase* create_extension(const InvDiffKinSolverParams &params);
+};
+/* END KinematicExtensionBuilder *******************************************************************************************/
+
+
+/* BEGIN KinematicExtensionNone ****************************************************************************************/
+class KinematicExtensionNone : public KinematicExtensionBase
+{
+    public:
+        KinematicExtensionNone(const InvDiffKinSolverParams &params)
+        : KinematicExtensionBase(params)
+        {
+            //nothing to do
+        }
+
+        ~KinematicExtensionNone() {}
+
+        virtual KDL::Jacobian adjust_jacobian(const KDL::Jacobian& jac_chain);
+        virtual void process_result_extension(const KDL::JntArray &q_dot_ik);
+};
+/* END KinematicExtensionNone **********************************************************************************************/
+
+
+/* BEGIN KinematicExtension6D ****************************************************************************************/
+class KinematicExtension6D : public KinematicExtensionBase
+{
+    public:
+        KinematicExtension6D(const InvDiffKinSolverParams &params)
+        : KinematicExtensionBase(params)
+        {
+            //nothing to do here
+        }
+
+        ~KinematicExtension6D() {}
+
+        virtual KDL::Jacobian adjust_jacobian(const KDL::Jacobian& jac_chain);
+        virtual void process_result_extension(const KDL::JntArray &q_dot_ik) = 0;
+        
+        KDL::Jacobian adjust_jacobian_6d(const KDL::Jacobian& jac_chain, const KDL::Frame full_frame, const KDL::Frame partial_frame, const ActiveCartesianDimension active_dim);
+};
+/* END KinematicExtension6D **********************************************************************************************/
+
+
+/* BEGIN KinematicExtensionBaseActive ****************************************************************************************/
+class KinematicExtensionBaseActive : public KinematicExtension6D
+{
+    public:
+        KinematicExtensionBaseActive(const InvDiffKinSolverParams &params)
+        : KinematicExtension6D(params)
+        {
+            base_vel_pub = nh_.advertise<geometry_msgs::Twist>("/base/twist_controller/command", 1);
+            
+            //ToDo: I don't like this hack! Can we pass the tf_listner_ somehow?
+            ///give tf_listener_ some time to fill buffer
+            ros::Duration(0.5).sleep();
+        }
+
+        ~KinematicExtensionBaseActive() {}
+
+        virtual KDL::Jacobian adjust_jacobian(const KDL::Jacobian& jac_chain);
```

Comment:
Method naming must be camelCased (http://wiki.ros.org/CppStyleGuide#Function_.2BAC8_Methods)